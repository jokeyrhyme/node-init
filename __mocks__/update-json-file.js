/* @flow */
'use strict';

let pkg /* : Object */ = {};

module.exports = async (
  pkgPath /* : string */,
  updater /* : (value: any) => Promise<any> */,
) /* : Promise<void> */ => {
  await updater(pkg);
};

module.exports.__setPackage = (p /* : Object */) => {
  pkg = p;
};
