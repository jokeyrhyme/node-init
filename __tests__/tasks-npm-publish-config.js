'use strict';

jest.mock('update-json-file');

const {
  lib: { packageJSON },
} = require('../lib/tasks/npm-publish-config.js');

test('packageJSON() fresh', async () => {
  const pkg = {};
  require('update-json-file').__setPackage(pkg);

  await packageJSON('some/path');

  expect(pkg).toMatchSnapshot();
});
