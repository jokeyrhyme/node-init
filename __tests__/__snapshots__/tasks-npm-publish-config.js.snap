// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`packageJSON() fresh 1`] = `
Object {
  "publishConfig": Object {
    "access": "public",
  },
}
`;
